From 5e40d1551af04499f6e0f408e07dc5046123c6c0 Mon Sep 17 00:00:00 2001
From: Miroslav Rezanina <mrezanin@redhat.com>
Date: Wed, 31 Jul 2013 08:02:36 +0200
Subject: pc: Drop superfluous RHEL-6 compat_props

Message-id: <1375201922-6794-4-git-send-email-armbru@redhat.com>
Patchwork-id: 52812
O-Subject: [RHEL-7 PATCH v3 03/15] pc: Drop superfluous RHEL-6 compat_props
Bugzilla: 983991
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Miroslav Rezanina <mrezanin@redhat.com>

Drop compatibility properties of

* devices we won't ship: ne2k_pci, pcnet, pc-sysfw, vmware-svga

* devices that are new in RHEL-7: nec-usb-xhci, ivshmem

Re ivshmem: actually, the current plan is not to ship it (bug 787463).
Executing the plan will only flip the reason for dropping its
compatibility property, so no code conflict here.

Signed-off-by: Markus Armbruster <armbru@redhat.com>

diff --git a/hw/i386/pc_piix.c b/hw/i386/pc_piix.c
index bdb1b11..96f6bf9 100644
--- a/hw/i386/pc_piix.c
+++ b/hw/i386/pc_piix.c
@@ -840,14 +840,6 @@ static QEMUMachine pc_machine_rhel700 = {
         .property = "romfile",\
         .value    = "pxe-e1000.rom",\
     },{\
-        .driver   = "ne2k_pci",\
-        .property = "romfile",\
-        .value    = "pxe-ne2k_pci.rom",\
-    },{\
-        .driver   = "pcnet",\
-        .property = "romfile",\
-        .value    = "pxe-pcnet.rom",\
-    },{\
         .driver   = "rtl8139",\
         .property = "romfile",\
         .value    = "pxe-rtl8139.rom",\
@@ -876,18 +868,6 @@ static QEMUMachine pc_machine_rhel700 = {
         .property = "autonegotiation",\
         .value    = "off",\
     },{\
-        .driver   = "nec-usb-xhci",\
-        .property = "msi",\
-        .value    = "off",\
-    },{\
-        .driver   = "nec-usb-xhci",\
-        .property = "msix",\
-        .value    = "off",\
-    },{\
-        .driver   = "ivshmem",\
-        .property = "use64",\
-        .value    = "0",\
-    },{\
         .driver   = "qxl",\
         .property = "revision",\
         .value    = stringify(3),\
@@ -912,10 +892,6 @@ static QEMUMachine pc_machine_rhel700 = {
         .property = "vgamem_mb",\
         .value    = stringify(8),\
     },{\
-        .driver   = "vmware-svga",\
-        .property = "vgamem_mb",\
-        .value    = stringify(8),\
-    },{\
         .driver   = "qxl-vga",\
         .property = "vgamem_mb",\
         .value    = stringify(8),\
@@ -928,10 +904,6 @@ static QEMUMachine pc_machine_rhel700 = {
         .property = "config-wce",\
         .value    = "off",\
     },{\
-        .driver   = "pc-sysfw",\
-        .property = "rom_only",\
-        .value    = stringify(1),\
-    },{\
         .driver   = TYPE_ISA_FDC,\
         .property = "check_media_rate",\
         .value    = "off",\
@@ -983,10 +955,6 @@ static QEMUMachine pc_machine_rhel700 = {
         .driver   = "VGA",\
         .property = "rombar",\
         .value    = stringify(0),\
-    },{\
-        .driver   = "vmware-svga",\
-        .property = "rombar",\
-        .value    = stringify(0),\
     }
 
 static void pc_init_rhel600(QEMUMachineInitArgs *args)
